lib/arjdbc/vertica/adapter.rb

```diff
@@ -279,21 +290,61 @@ def vertica_random_temp_table_name
       "temporary_table_#{::SecureRandom.hex}"
     end
 
-    def type_to_sql(type, limit = nil, precision = nil, scale = nil) #:nodoc:
-      super unless (type.to_sym == :integer || type.to_sym == :bigserial)
+  end
+end
 
-      if native = native_database_types[type.to_sym]
-        (native.is_a?(Hash) ? native[:name] : native).dup
-      else
-        type.to_s
+module ActiveRecord
```

Comment:
So after looking at how the Postgresql jdbc adapter does this

https://github.com/jruby/activerecord-jdbc-adapter/blob/master/lib/arjdbc/postgresql/adapter.rb#L13

overriding this method isn't crazy, however i don't know if this is the correct place in the code to put it.

the entire purpose of this overriding is to have the logic
`(type != :primary_key && type != :integer)  && (limit ||= native.is_a?(Hash) && native[:limit])`

which prevents the code from converting `t.integer, limit: 8` into `"integer(8)"`

Reply:
```
module ConnectionAdapters
  class AbstractAdapter
    class SchemaCreation
      include ::ArJdbc::Vertica

      def type_to_sql(type, limit: nil, precision: nil, scale: nil, **) # :nodoc:
        type = type.to_sym if type

        if native = native_database_types[type]
          column_type_sql = (native.is_a?(Hash) ? native[:name] : native).dup
          if (type != :primary_key && type != :integer)  && (limit ||= native.is_a?(Hash) && native[:limit])
            column_type_sql << "(#{limit})"
            return
          end
        end

        super
      end

    end
  end
end
```

Is it worth reducing the required logic to something like ^^ ? Not sure if it is, just asking the question.

Reply:
so if i'm understanding correctly, basically we write one override `if` block and then let `super` handle everything else?

Reply:
If you think it's helpful. Otherwise I'm cool with the copy.

Reply:
🙄 

`NoMethodError: super: no superclass method `type_to_sql' for #<ActiveRecord::ConnectionAdapters::AbstractAdapter::SchemaCreation:0xf9879ac>`

Reply:
At this point i prefer just having the full copy code here. Getting it to be able to call super for condensation of code is turning out to be more work than i think it will be worth.